File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/registry/RegistryPayMethod.java
package com.esferalia.aon.occam.api.model.registry;

import java.io.Serializable;

import com.esferalia.aon.occam.api.model.finance.PayMethod;

public class RegistryPayMethod implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private Integer id;
	private Integer domain;
	private Integer registry;
	private PayMethod payMethod;
	private RegistryBank rbank;
	private Short numberOfPymnts;
	private Short daysToFirstPymnt;
	private Short daysBetweenPymnts;
	private String pymntDays;
	
	private boolean dirty;
	private boolean removed;
	
	public RegistryPayMethod() {
		this.numberOfPymnts = 1;
		this.daysToFirstPymnt = 0;
		this.daysBetweenPymnts = 0;
		this.pymntDays = "";
	}
	
	public Integer getId() {
		return id;
	}
	
	public RegistryPayMethod setId(Integer id) {
		this.id = id;
		return this;
	}
	
	public Integer getDomain() {
		return domain;
	}
	
	public RegistryPayMethod setDomain(Integer domain) {
		setDirty(true);
		this.domain = domain;
		return this;
	}
	
	public Integer getRegistry() {
		return registry;
	}
	
	public RegistryPayMethod setRegistry(Integer registry) {
		setDirty(true);
		this.registry = registry;
		return this;
	}
	
	public PayMethod getPayMethod() {
	    if(payMethod == null) {
	        payMethod = new PayMethod();
	    }
		return payMethod;
	}
	
	public RegistryPayMethod setPayMethod(PayMethod payMethod) {
		setDirty(true);
		this.payMethod = payMethod;
		return this;
	}
	
	public RegistryBank getRbank() {
	    if(rbank == null) {
	        rbank = new RegistryBank();
	    }
		return rbank;
	}
	
	public RegistryPayMethod setRbank(RegistryBank rbank) {
		setDirty(true);
		this.rbank = rbank;
		return this;
	}
	
	public Short getNumberOfPymnts() {
		if(numberOfPymnts == null) {
			numberOfPymnts = 1;
		}
		return numberOfPymnts;
	}
	
	public RegistryPayMethod setNumberOfPymnts(Short numberOfPymnts) {
		setDirty(true);
		this.numberOfPymnts = numberOfPymnts;
		return this;
	}
	
	public Short getDaysToFirstPymnt() {
		if(daysToFirstPymnt == null) {
			daysToFirstPymnt = 0;
		}
		return daysToFirstPymnt;
	}
	
	public RegistryPayMethod setDaysToFirstPymnt(Short daysToFirstPymnt) {
		setDirty(true);
		this.daysToFirstPymnt = daysToFirstPymnt;
		return this;
	}
	
	public Short getDaysBetwenPymnts() {
		if(daysBetweenPymnts == null) {
			daysBetweenPymnts = 0;
		}
		return daysBetweenPymnts;
	}
	
	public RegistryPayMethod setDaysBetwenPymnts(Short daysBetweenPymnts) {
		setDirty(true);
		this.daysBetweenPymnts = daysBetweenPymnts;
		return this;
	}
	
	public String getPymntDays() {
		if(pymntDays == null) {
			pymntDays = "";
		}
		return pymntDays;
	}
	
	public RegistryPayMethod setPymntDays(String pymntDays) {
		setDirty(true);
		this.pymntDays = pymntDays;
		return this;
	}
	
	public boolean isDirty() {
		return dirty;
	}
	
	public RegistryPayMethod setDirty(boolean dirty) {
		this.dirty = dirty;
		return this;
	}
	
	public boolean isRemoved() {
		return removed;
	}
	
	public RegistryPayMethod setRemoved(boolean removed) {
		this.removed = removed;
		return this;
	}
	
	public void remove() {
		setRemoved(true);
	}
	
}

File: aon-web-aio/src/main/webapp/models/RegistryPaymethod.js
import * as LS from '../services/localStorageService.js';

export class RegistryPaymethod {

    id;
    domain;
    registry;
    paymethod;
    bank;
    numberOfPymnts;
    daysToFirstPymnt;
    daysBetweenPymnts;
    pymntDays;

    dirty;
    removed;

    constructor(rpaymethod) {
        if(rpaymethod) {
            this.id = rpaymethod.id;
            this.domain = rpaymethod.domain || LS.getDomainId();
            this.registry = rpaymethod.registry;
            this.paymethod = rpaymethod.paymethod;
            this.bank = rpaymethod.bank;
            this.numberOfPymnts = rpaymethod.numberOfPymnts || 1;
            this.daysToFirstPymnt = rpaymethod.daysToFirstPymnt || 0;
            this.daysBetweenPymnts = rpaymethod.daysBetweenPymnts || 0;
            this.pymntDays = rpaymethod.pymntDays || '';
            this.dirty = rpaymethod.dirty || false;
            this.removed = rpaymethod.removed || false;
        } else {
            this.domain = LS.getDomainId();
            this.numberOfPymnts = 1;
            this.daysToFirstPymnt = 0;
            this.daysBetweenPymnts = 0;
            this.pymntDays = '';
            this.dirty = false;
            this.removed = false;
        }
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
        return this;
    }

    getDomain() {
        return this.domain;
    }

    setDomain(domain) {
        this.setDirty(true);
        this.domain = domain;
        return this;
    }

    getRegistry(){
        return this.registry;
    }

    setRegistry(registry){
        this.setDirty(true);
        this.registry = registry;
        return this;
    }

    getPaymethod() {
        return this.paymethod;
    }

    setPaymethod(paymethod) {
        this.setDirty(true);
        this.paymethod = paymethod;
        return this;
    }

    getBank() {
        return this.bank;
    }

    setBank(bank) {
        this.setDirty(true);
        this.bank = bank;
        return this;
    }

    getNumberOfPymnts() {
        return this.numberOfPymnts;
    }

    setNumberOfPymnts(numberOfPymnts) {
        this.setDirty(true);
        this.numberOfPymnts = numberOfPymnts;
        return this;
    }

    getDaysToFirstPymnt(){
        return this.daysToFirstPymnt;
    }

    setDaysToFirstPymnt(daysToFirstPymnt) {
        this.setDirty(true);
        this.daysToFirstPymnt = daysToFirstPymnt;
        return this;
    }

    getDaysBetweenPymnts(){
        return this.daysBetweenPymnts;
    }

    setDaysBetweenPymnts(daysBetweenPymnts) {
        this.setDirty(true);
        this.daysBetweenPymnts = daysBetweenPymnts;
        return this;
    }

    getPymntDays() {
        return this.pymntDays;
    }

    setPymntDay(pymntDays) {
        this.setDirty(true);
        this.pymntDays = pymntDays;
        return this;
    }

    isDirty() {
        return this.dirty;
    }

    setDirty(dirty) {
        this.dirty = dirty;
        return this;
    }

    isRemoved() {
        return this.removed;
    }

    setRemoved(removed) {
        this.removed = removed;
        return this;
    }

    remove() {
        this.setRemoved(true);
    }
} 
